### PR TITLE
COG-30 CAP-API Retrieve a list of permissions

### PR DESCRIPTION
**Pull request by vladimirPe**
### JIRA Issue Link
- [COG-30: Name of Story](https://osi-cwds.atlassian.net/browse/COG-30)

### Technical Description
Permissions RestAPI endpoint was added 

### Tests
- [x] I have included unit tests 
- [ ] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Configuration changes
N/A
### Technical debt
N/A

### CODE REVIEW

File: src/main/java/gov/ca/cwds/idm/persistence/PermissionRepository.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.List;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+
+public interface PermissionRepository extends CrudRepository<Permission, String> {
+
+  @Query("SELECT p.name FROM Permission p order by p.name asc")
+  @Cacheable("permissionNames")
```
Comment (DmitriyRudenko) [inline src/main/java/gov/ca/cwds/idm/persistence/PermissionRepository.java]: 1. Spring doesn't recommend to use @Cachable etc on interfaces( we use "proxy target class" proxy strategy which may be a problem for caching)
2. Permission repo is read only, so I would recommend to use ReadOnlyRepository as base for PermissionRepository
3. We have jpa caching as well which is enough

Summary:
Just remove Spring caching annotations and use gov.ca.cwds.data.auth.ReadOnlyRepository as base

**Comment by DmitriyRudenko** [comment]
please run integration tests both for dev and cognito modes, docs are here: 

https://github.com/ca-cwds/perry/tree/master/integration-tests

File: src/main/java/gov/ca/cwds/idm/persistence/PermissionRepository.java (hunk 1 above)
Comment (vladimirPe) [reply src/main/java/gov/ca/cwds/idm/persistence/PermissionRepository.java]: good point

**Comment by vladimirPe** [comment]
jenkins retest it please

**Comment by vladimirPe** [comment]
retest this please

**Comment by DmitriyRudenko** [comment]
Retest this please

**Comment by DmitriyRudenko** [comment]
retest this please

**Pull request merged**